pyproject.toml

```diff
@@ -28,41 +28,153 @@ classifiers = [
     "Natural Language :: English",
     "Programming Language :: C++",
     "Programming Language :: Python",
-    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
     "Topic :: Scientific/Engineering",
     "Topic :: Scientific/Engineering :: Mathematics",
     "Topic :: Scientific/Engineering :: Physics",
 ]
 dependencies = [
     "importlib-resources; python_version<'3.9'",
     "numpy",
-    "platformdirs>=2.2.0",
+    "platformdirs>=2.2",
     "pytools>=2024.1.5",
 ]
 
 [project.optional-dependencies]
 oclgrind = [
-    "oclgrind_binary_distribution>=18.3",
+    "oclgrind-binary-distribution>=18.3",
 ]
 pocl = [
-    "pocl_binary_distribution>=1.2",
+    "pocl-binary-distribution>=1.2",
 ]
 test = [
-    "Mako",
-    "pytest>=7.0.0",
+    "flake8",
+    "flake8-bugbear",
+    "flake8-comprehensions",
+    "flake8-isort",
+    "flake8-quotes",
+    "isort",
+    "mako",
+    "mypy",
+    "pep8-naming",
+    "pylint",
+    "pytest>=7",
 ]
 
 [project.urls]
-Homepage = "https://mathema.tician.de/software/pyopencl"
 Documentation = "https://documen.tician.de/pyopencl"
+Homepage = "https://mathema.tician.de/software/pyopencl"
 Repository = "https://github.com/inducer/pyopencl"
 
+[tool.scikit-build]
+sdist.exclude = [
+    ".mypy_cache",
+    ".ci",
+    ".github",
+    ".conda-ci-build-configure.sh",
+    "doc/upload-docs.sh",
+    ".editorconfig",
+    "TODOs",
+    "run-*.sh",
+    "setup.cfg",
+]
+
+[tool.aksetup]
+disable-editable-pip-install = true
+enable-flake8-isort = true
+enable-flake8-bugbear = true
```

Comment:
@inducer What do you think about something like this instead of the comments that were in the `setup.cfg` file? If nothing else, it makes them easy to parse going forward :grin:

Reply:
That's obviously nicer, let's do that. Maybe `inducer-ci-support` as the tool name? `aksetup` was the nickname for that hodgepodge of setuptools hacks that we're killing.

I couldn't help myself: here's a one-liner dotted-name-retriever for TOML:
```sh
python3 -c 'import toml, sys; data = toml.load(sys.argv[1]); [data := data[name] for name in sys.argv[2].split(".")]; print(data)' src/pyopencl/pyproject.toml project.readme
README.rst
```
Then we can get rid of the greppy stuff (mostly), while deprecating support for `setup.cfg`.

Reply:
That looks very nice! I'll go ahead and rename it.

Reply:
Done in https://github.com/inducer/pyopencl/compare/73ca584e3ab8186873d91d70d50a3f4cffcab6b5..e7b9adc6389d34d5cb9d72e848024456782ac7fc.